src/riak_kv_vnode.erl

```diff
@@ -1540,26 +1540,19 @@ prepare_put_existing_object(#state{idx =Idx} = State,
     {ActorId, State2} = maybe_new_key_epoch(Coord, State, OldObj, RObj),
     case put_merge(Coord, LWW, OldObj, RObj, ActorId, StartTime) of
         {oldobj, OldObj} ->
-            {{false, OldObj}, PutArgs, State2};
+            {{false, {OldObj, no_old_object}}, PutArgs, State2};
```

Comment:
True, but I really felt like having the _same structure_ for all cases actually made the code easier to read, and also to _know_ that you've taken the old object stuff into account in each code path, which is why I pushed it through to all the cases.